src/Files.App/Actions/DebounceAction.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Files.App.Actions
+{
+	public abstract class DebouncedAction : IAction
+	{
+		private DateTime lastExecuted = DateTime.MinValue;
+		private readonly TimeSpan debounceTime;
+
+		protected DebouncedAction(TimeSpan debounceDuration)
+		{
+			debounceTime = debounceDuration;
+		}
+
+		public virtual bool CanExecuteNow()
+		{
+			return DateTime.Now - lastExecuted > debounceTime;
+		}
+
+		public virtual void MarkAsExecuted()
+		{
+			lastExecuted = DateTime.Now;
+		}
```

Comment:
The name is not exactly telling of what it's doing. It should be something like `MarkLastExecutionTime`.